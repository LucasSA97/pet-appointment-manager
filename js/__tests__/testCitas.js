import Citas from "../classes/citas";

describe("Probar la clase Citas", () => {
    const citas = new Citas
    const id = Date.now()

    test('Agregar una nueva cita ', () => { 
        const citaObjeto = {
            id,
            mascota: 'Poli',
            propietario: 'Lucas',
            telefono: '3241315',
            fecha: '02/10/2023',
            hora: '10:30',
            sintomas: 'Solo duerme'
        } 

        citas.agregarCita(citaObjeto)
        expect(citas).toMatchSnapshot()
     })

     test('Actualizar cita', () => {
        const citaActualizada = {
            mascota: 'Ema',
            id,
            propietario: 'Lucas',
            telefono: '3241315',
            fecha: '02/10/2023',
            hora: '10:30',
            sintomas: 'Solo come'
        } 
        citas.editarCita(citaActualizada);
        expect(citas).toMatchSnapshot()
     })

     test('Eliminar cita', () => {
 
        citas.eliminarCita(id);
        expect(citas).toMatchSnapshot()
     })
})