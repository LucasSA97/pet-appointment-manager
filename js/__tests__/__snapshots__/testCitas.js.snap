// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase Citas Actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "02/10/2023",
      "hora": "10:30",
      "id": 1702651200895,
      "mascota": "Ema",
      "propietario": "Lucas",
      "sintomas": "Solo come",
      "telefono": "3241315",
    },
  ],
}
`;

exports[`Probar la clase Citas Agregar una nueva cita  1`] = `
Citas {
  "citas": [
    {
      "fecha": "02/10/2023",
      "hora": "10:30",
      "id": 1702651200895,
      "mascota": "Poli",
      "propietario": "Lucas",
      "sintomas": "Solo duerme",
      "telefono": "3241315",
    },
  ],
}
`;

exports[`Probar la clase Citas Eliminar cita 1`] = `
Citas {
  "citas": [],
}
`;
